torch/fx/experimental/symbolic_shapes.py

```diff
@@ -3183,7 +3184,12 @@ def _maybe_guard_eq(self, expr: Union["sympy.Eq", "sympy.Ne"], concrete_bool: bo
                 solutions = sympy.solve(lhs - rhs, free[0], dict=True)
                 if len(solutions) != 1:
                     return
+                # TODO: remove after checking our 'solve' is equivalent.
                 solution = solutions[0][free[0]]
+                r = try_solve(expr, free[0])
+                if r is None or r[1] != solution:
+                    rstr = "None" if r is None else r[1]
+                    print(f"EQ ({free[0]}): {lhs} = {rhs} ==> {solution} != {rstr} (ours)")
```

Comment:
Left these prints so that I can see cases where `sympy.solve` works but our doesn't.